flake.nix

```diff
@@ -1,77 +1,144 @@
+            pkgs.runCommand "hc-scaffold"
+              {
+                buildInputs = [ pkgs.makeWrapper ];
+                src = customTemplatePath;
+              } ''
+                mkdir $out
+                mkdir $out/bin
+                # We create the bin folder ourselves and link every binary in it
+                ln -s ${scaffolding}/bin/* $out/bin
+                # Except the hello binary
+                rm $out/bin/hc-scaffold
+                cp $src -R $out/template
+                # Because we create this ourself, by creating a wrapper
+                makeWrapper ${scaffolding}/bin/hc-scaffold $out/bin/hc-scaffold \
+                  --add-flags "--template $out/template"
+              	'';
         };
+        systems = builtins.attrNames inputs.holonix.devShells;
+        perSystem = { inputs', self', config, system, pkgs, lib, ... }: {
+          formatter = pkgs.nixpkgs-fmt;
 
-        # TODO: Expose the scaffolding tool CLI as the main package for this crate
+          packages.hc-scaffold =
+            let
+              pkgs = import nixpkgs {
+                inherit system;
+                overlays = [ (import rust-overlay) ];
+              };
+              rustToolchain = pkgs.rust-bin.stable."1.79.0".minimal;
+              craneLib = (crane.mkLib pkgs).overrideToolchain rustToolchain;
+              crateInfo = craneLib.crateNameFromCargoToml { cargoToml = ./Cargo.toml; };
 
-        checks.custom-template = flake.lib.wrapCustomTemplate {
-          inherit pkgs system;
-          customTemplatePath = ./templates/custom-template/custom-template;
-        };
+              # source filtering to ensure builds using include_str! or include_bytes! succeed
+              # https://crane.dev/faq/building-with-non-rust-includes.html
+              excludedDirs = [
+                ".git"
+                "target"
+                "result"
+                # Add more directories to exclude here
+              ];
+              scaffoldFilter = path: type:
+                let
+                  baseName = baseNameOf path;
+                  isExcluded = builtins.elem baseName excludedDirs;
+                in
+                  !(type == "directory" && isExcluded);
+            in
+            craneLib.buildPackage {
+              pname = "hc-scaffold";
+              version = crateInfo.version;
+              src = lib.cleanSourceWith {
+                src = ./.;
+                filter = scaffoldFilter;
+                name = "source";
+              };
+              doCheck = false;
+
+              buildInputs = [ pkgs.openssl pkgs.go ]
+                ++ (lib.optionals pkgs.stdenv.isDarwin
+                (with pkgs.darwin.apple_sdk.frameworks; [
+                  CoreFoundation
+                  SystemConfiguration
+                  Security
+                ]));
 
+              nativeBuildInputs = [ pkgs.perl ];
+              env = {
+                OPENSSL_DIR = "${pkgs.openssl.dev}";
```

Comment:
Does this fail to build without openssl? I'd like to take a look at why with you if possible. Probably not right now because I'm busy with some Holochain work but I'd managed to eliminate the need for this dependency and I'd like to figure out why it's needed again! :)

Reply:
Seems to work now without openssl, but I need to include `perl`

```nix
 nativeBuildInputs = [ pkgs.perl ];
```